src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Resources/ManagedImagesMonikers.cs

```diff
@@ -1,40 +1,127 @@
-        public static ImageMoniker SharedProjectWarning => new ImageMoniker { Guid = s_manifestGuid, Id = 20 };
-        public static ImageMoniker WarningSmall => new ImageMoniker { Guid = s_manifestGuid, Id = 21 };
+        public static ImageMoniker Component              => new ImageMoniker { Guid = s_manifestGuid, Id = 4 };
+        public static ImageMoniker ComponentPrivate       => new ImageMoniker { Guid = s_manifestGuid, Id = 5 };
+        public static ImageMoniker ComponentWarning       => new ImageMoniker { Guid = s_manifestGuid, Id = 6 };
+        public static ImageMoniker ErrorSmall             => new ImageMoniker { Guid = s_manifestGuid, Id = 7 };
+        public static ImageMoniker LibraryWarning         => new ImageMoniker { Guid = s_manifestGuid, Id = 8 };
+        public static ImageMoniker NuGetGrey              => new ImageMoniker { Guid = s_manifestGuid, Id = 9 };
+        public static ImageMoniker NuGetGreyPrivate       => new ImageMoniker { Guid = s_manifestGuid, Id = 10 };
+        public static ImageMoniker NuGetGreyWarning       => new ImageMoniker { Guid = s_manifestGuid, Id = 11 };
+        public static ImageMoniker ReferenceGroup         => new ImageMoniker { Guid = s_manifestGuid, Id = 12 };
+        public static ImageMoniker ReferenceGroupWarning  => new ImageMoniker { Guid = s_manifestGuid, Id = 13 };
+        public static ImageMoniker ReferencePrivate       => new ImageMoniker { Guid = s_manifestGuid, Id = 14 };
+        public static ImageMoniker Sdk                    => new ImageMoniker { Guid = s_manifestGuid, Id = 15 };
+        public static ImageMoniker SdkPrivate             => new ImageMoniker { Guid = s_manifestGuid, Id = 16 };
+        public static ImageMoniker SdkWarning             => new ImageMoniker { Guid = s_manifestGuid, Id = 17 };
+        public static ImageMoniker SharedProject          => new ImageMoniker { Guid = s_manifestGuid, Id = 18 };
+        public static ImageMoniker SharedProjectPrivate   => new ImageMoniker { Guid = s_manifestGuid, Id = 19 };
+        public static ImageMoniker SharedProjectWarning   => new ImageMoniker { Guid = s_manifestGuid, Id = 20 };
+        public static ImageMoniker WarningSmall           => new ImageMoniker { Guid = s_manifestGuid, Id = 21 };
+
+        #region DebuggerDisplay support for known project system image monikers
+
+        // These methods are called by the debugger, as instructed by the DebuggerDisplayAttributes at the top of the file.
+
+        internal static string ImageMonikerDebugDisplay(ImageMoniker moniker) => DebugDisplay(moniker.Guid, moniker.Id);
+
+        internal static string ProjectImageMonikerDebugDisplay(ProjectImageMoniker moniker) => DebugDisplay(moniker.Guid, moniker.Id);
+
+        private static string DebugDisplay(Guid guid, int id)
+        {
+            if (guid == s_manifestGuid)
+            {
+                switch (id)
+                {
+                    case 0:  return nameof(ApplicationPrivate);
```

Comment:
An alternative for this would to be checking against the static fields themselves to avoid getting out of sync with the id.

Reply:
I agree it'd be good to prevent them getting out of sync.

`case` needs a compile time constant. We'd have to pull out consts for the IDs. Do you think it'd be worth it?

Another alternative would be use reflection and a loop. Not sure if that overhead would adversely impact the debugging experience. It feels a bit dirty.

Reply:
I thought pattern matching might help here?

Reply:
I'll try it out. Pattern matching falls back to chained if/else rather than a jump table. Might not be a big deal. Still orders of magnitude better than reflection.

Reply:
Pattern matching didn't work. It might in a future version of C#. Our options are to leave it or pull out constants. Or add some documentation saying "if you change these, update this over here too".